sdk/core/Azure.Core.Experimental/tests/Variant/VariantUsage.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using NUnit.Framework;
+
+namespace Azure.Core.Experimental.Tests
+{
+    public class VariantUsage
+    {
+        [Test]
+        public void CanUseVariantAsString()
+        {
+            Variant variant = "hi";
+
+            Assert.AreEqual("hi", (string)variant);
+
+            Assert.IsTrue("hi" == (string)variant);
+            Assert.IsTrue((string)variant == "hi");
+
+            Assert.AreEqual("hi", $"{(string)variant}");
+        }
+
+        [Test]
+        public void CanTestForNull()
+        {
+            Variant variant = Variant.Null;
```

Comment:
nit: this depends on the implementation of Variant.Null. Should we also add a test like:

```c#
Assert.True(Variant.IsNull(new Variant((object?)null));
```

Reply:
Yep, I like it - thank you!